components/style/properties/gecko.mako.rs

```diff
@@ -1323,6 +1323,48 @@ pub fn clone_transform_from_list(
     }
 </%def>
 
+<%def name="impl_transform_origin(ident, gecko_ffi_name)">
+    #[allow(non_snake_case)]
+    pub fn set_${ident}(&mut self, v: values::computed::TransformOrigin) {
+        self.gecko.${gecko_ffi_name}[0].set(v.horizontal);
+        self.gecko.${gecko_ffi_name}[1].set(v.vertical);
+        if self.gecko.${gecko_ffi_name}.len() == 3 {
+            self.gecko.${gecko_ffi_name}[2].set(v.depth);
+        }
+    }
+
+    #[allow(non_snake_case)]
+    pub fn copy_${ident}_from(&mut self, other: &Self) {
+        self.gecko.${gecko_ffi_name}[0].copy_from(&other.gecko.${gecko_ffi_name}[0]);
+        self.gecko.${gecko_ffi_name}[1].copy_from(&other.gecko.${gecko_ffi_name}[1]);
+        if self.gecko.${gecko_ffi_name}.len() == 3 {
```

Comment:
nit: This (and above) probably deserve a comment. Mind leaving one?

Also, does this have the warning issue? Should we address it like I suggested in https://github.com/rust-lang/rust/issues/45850? Maybe at least reference that?

Reply:
I'm... not really happy with adding those boilerplates just to workaround a non-blocking rustc issue :/

Reply:
I don't care that much, but seeing warnings on the build like that is somewhat scary, so I think at least a comment would be on point.